tests/core/pyspec/eth2spec/test/phase0/fork_choice/test_on_block.py

```diff
@@ -703,3 +703,29 @@ def test_new_finalized_slot_is_justified_checkpoint_ancestor(spec, state):
     assert store.justified_checkpoint == another_state.current_justified_checkpoint
 
     yield 'steps', test_steps
+
+
+@with_all_phases
+@spec_state_test
+def test_proposer_score_boost_same_slot_untimely_block(spec, state):
```

Comment:
Suggest doing this same test but for `store.genesis_time + block.slot * spec.config.SECONDS_PER_SLOT +
                 spec.config.SECONDS_PER_SLOT // spec.INTERVALS_PER_SLOT - 1` and showing it does mutate